### ISSUE TITLE
Looks for type def file for the names of paths in my project root

### ISSUE THREAD

**Issue by kylegoetz** [TypeStrong/ts-node#725]
I have a project structure:

    <project root>
    -- tsconfig.json
    -- src/
    -- tools/
      init.ts
    -- node_modules/
    -- .git/
    -- .github/
    -- tools/

and so forth. When I run `ts-node tools/init.ts` I get the following errors:

    /Users/kylegoetz/Documents/workspace/test-scaffold/node_modules/ts-node/src/index.ts:261
        return new TSError(diagnosticText, diagnosticCodes)
               ^
    TSError: ⨯ Unable to compile TypeScript:
    error TS2688: Cannot find type definition file for '.git'.
    error TS2688: Cannot find type definition file for '.github'.
    error TS2688: Cannot find type definition file for 'node_modules'.
    error TS2688: Cannot find type definition file for 'scripts'.
    error TS2688: Cannot find type definition file for 'types'.

Why is ts-node looking in my project's root for type definition files for these folders?

**Comment by blakeembrey** [comment]
You didn’t specify your `tsconfig.json` via project. It doesn’t find it via the first file name, it’s based on current working directory.

**Comment by kylegoetz** [comment]
Can you re-open this? The diagnosis was not correct.

I have a tsconfig.json in the CWD from which I call ts-node. But to double check, I made it explicit and the exact same result comes: `./node_modules/.bin/ts-node --project tsconfig.json tools/init.ts`. Same exact result.

For reference, here is my tsconfig:

    {
      "compilerOptions": {
        "moduleResolution": "node",
        "target": "ESNext",
        "module":"es2015",
        "lib": ["es2015", "es2016", "es2017"],
        "strict": true,
        "sourceMap": true,
        "declaration": true,
        "allowSyntheticDefaultImports": true,
        "experimentalDecorators": true,
        "emitDecoratorMetadata": true,
        "resolveJsonModule": true,
        "declarationDir": "dist/types",
        "outDir": "dist/lib",
        "typeRoots": [
          "./node_modules/@types",
          "../DefinitelyTyped",
          "types"
        ]
      },
      "include": [
        "src/**/*"
      ]
    }

I understand "include" is ignored by ts-node, and I've changed it to a few values to be sure.

typeRoots is correct. From the CWD, there is a types path. From the CWD, the parent directory has a git clone of the DefinitelyTyped repo. I also have some installed via npm locally to the project, hence the ./node_modules/@types entry.

Any ideas?

**Comment by kylegoetz** [comment]
Also, the full output:

`Kyles-MBP:test-scaffold kylegoetz$ ./node_modules/.bin/ts-node --project tsconfig.json tools/init.ts 

/Users/kylegoetz/Documents/workspace/test-scaffold/node_modules/ts-node/src/index.ts:261
    return new TSError(diagnosticText, diagnosticCodes)
           ^
TSError: ⨯ Unable to compile TypeScript:
error TS2688: Cannot find type definition file for '.git'.
error TS2688: Cannot find type definition file for '.github'.
error TS2688: Cannot find type definition file for 'node_modules'.
error TS2688: Cannot find type definition file for 'scripts'.
error TS2688: Cannot find type definition file for 'types'.

    at createTSError (/Users/kylegoetz/Documents/workspace/test-scaffold/node_modules/ts-node/src/index.ts:261:12)
    at getOutput (/Users/kylegoetz/Documents/workspace/test-scaffold/node_modules/ts-node/src/index.ts:367:40)
    at Object.compile (/Users/kylegoetz/Documents/workspace/test-scaffold/node_modules/ts-node/src/index.ts:558:11)
    at Module.m._compile (/Users/kylegoetz/Documents/workspace/test-scaffold/node_modules/ts-node/src/index.ts:439:43)
    at Module._extensions..js (module.js:671:10)
    at Object.require.extensions.(anonymous function) [as .ts] (/Users/kylegoetz/Documents/workspace/test-scaffold/node_modules/ts-node/src/index.ts:442:12)
    at Module.load (module.js:573:32)
    at tryModuleLoad (module.js:513:12)
    at Function.Module._load (module.js:505:3)
    at Function.Module.runMain (module.js:701:10)
`

Very bizarre is that the `/Users/kylegoetz/Documents/workspace/test-scaffold/node_modules/ts-node/src/index.ts:442:12` referenced in the trace doesn't actually exist. There isn't even a "src" directory, only "dist"

**Comment by blakeembrey** [comment]
You’re entirely right, misread the initial description. I’ve never seen this before, can you provide a reproduction or some more information on the specific versions you’re using?

**Comment by kylegoetz** [comment]
Cannot for the life of you tell you what the problem ended up being. Through sheer dumb luck I ended up with code that works. I did have to eventually add `-O '{"module":"commonjs"}'` so I could compile all my code that uses `import` statements, but that wasn't related to the problem. I have no earthly idea. Sorry I couldn't help you find a bug. Essentially I commented out everything, uncommented one block at a time and ran ts-node, and eventually I had uncommented everything and it still worked.
